mlir/lib/Dialect/MemRef/IR/MemRefMemorySlot.cpp

```diff
@@ -187,14 +187,28 @@ DeletionKind memref::LoadOp::removeBlockingUses(
   return DeletionKind::Delete;
 }
 
-/// Returns the index of a memref in attribute form, given its indices.
+/// Returns the index of a memref in attribute form, given its indices. Returns
+/// a null pointer if whether the indices form a valid index for the provided
+/// MemRefType cannot be computed.
 static Attribute getAttributeIndexFromIndexOperands(MLIRContext *ctx,
-                                                    ValueRange indices) {
+                                                    ValueRange indices,
+                                                    MemRefType memrefType) {
+  ArrayRef<int64_t> shape = memrefType.getShape();
+  if (indices.size() != memrefType.getShape().size())
+    return {};
```

Comment:
Isn't this covered by the verifier?

Reply:
Yes, it's just a matter of whether or not we assume `getAttributeIndexFromIndexOperands` takes operands from LoadOp and StoreOp. This is probably a safe assumption, if documented.